packaging/opensuse/snapd.spec

```diff
@@ -203,40 +199,41 @@ export LDFLAGS
 # Generate autotools build system files.
 pushd %{indigo_srcdir}/cmd
 autoreconf -i -f
+
 %configure \
     %{!?with_apparmor:--disable-apparmor} \
     --libexecdir=%{_libexecdir}/snapd \
     --enable-nvidia-biarch \
     %{?with_multilib:--with-32bit-libdir=%{_prefix}/lib} \
     --with-snap-mount-dir=%{snap_mount_dir} \
     --enable-merged-usr
+
 popd
 
 %build
-# Build golang executables, with the following exceptions everything is built the same way:
-# - snap-exec and snap-update-ns is built statically.
-# - snapd has a variant that uses test keys instead of production keys.
+%make_build -C %{indigo_srcdir}/cmd
+# Use the common packaging helper for building.
 #
 # NOTE: indigo_gopath takes priority over GOPATH. This ensures that we
 # build the code that we intended in case GOPATH points to another copy.
-GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie %{import_path}/cmd/snap
-GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie %{import_path}/cmd/snapctl
-GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie %{import_path}/cmd/snap-seccomp
-GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=default -ldflags '-extldflags "-static"' %{import_path}/cmd/snap-update-ns
-GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=default -ldflags '-extldflags "-static"' %{import_path}/cmd/snap-exec
-%if 0%{?with_test_keys}
-GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie -tags withtestkeys %{import_path}/cmd/snapd
-%else
-GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie %{import_path}/cmd/snapd
-%endif
-
-# Build C executables
-%make_build -C %{indigo_srcdir}/cmd
+%make_build -f %{indigo_srcdir}/packaging/snapd.mk \
+	with_core_bits=0 \
+	with_alt_snap_mount_dir=%{!?with_alt_snap_mount_dir:0}%{?with_alt_snap_mount_dir:1} \
+	with_apparmor=%{!?with_apparmor:0}%{?with_apparmor:1} \
+	with_test_keys=%{!?with_test_keys:0}%{?with_test_keys:1} \
+	GOPATH=%{indigo_gopath}:$GOPATH \
+	all
 
 %check
-# Run tests with fixed locale that is expected by unicode/color code.
-LC_ALL=C.UTF-8 GOPATH=%{indigo_gopath}:$GOPATH go test %{import_path}/...
 %make_build -C %{indigo_srcdir}/cmd check
+# Use the common packaging helper for testing.
+%make_build -f %{indigo_srcdir}/packaging/snapd.mk \
+	with_core_bits=0 \
```

Comment:
With #4880 there was no need to repeat this all the time.

Reply:
The need to repeat could go away with a preamble.mk being generated by the spec file.

I didn't want the makefile to make any assumptions because a new variable being introduced would silently change the semantics of old packaging. I want this to fail early and loudly.

Reply:
Note that the scope of this PR is smaller than a main central makefile.

Reply:
This is now improved :)